### ISSUE TITLE
[a11y]: TagSet expansion and collapse is not conveyed to screen readers

### ISSUE THREAD

**Issue by elycheea** [carbon-design-system/ibm-products#8050]
### Package

@carbon/ibm-products

### Package version

v2.72.1

### Automated testing tool and ruleset

manual

### Assistive technology

JAWS

### Description

Reported by Raghu during screen reader testing (both PageHeader and TagSet).

The `+n` button which extends the tag list — acts as expand/collapse — however, the status is not conveyed to screen reader users.

#### Recommended fix
Use ARIA expand/collapse properties to convey the open or closed status to screen reader users.

#### User impact
When the name, role, or state of an element is not conveyed appropriately, it prevents screen reader users from interacting smoothly with the page element.

### WCAG 2.1 Violation

4.1.2 Name, Role, Value 

### Reproduction/example

https://ibm-products.carbondesignsystem.com/?path=/story/components-tagset--many-tags

### Steps to reproduce

1. Navigate to a TagSet story with the overflow tooltip.
2. Use a screen reader and open the tooltip.

Expansion and collapse of the menu is not announced[^1].

[^1]: Compare this to the [OperationalTag](https://react.carbondesignsystem.com/?path=/story/components-tag--operational) implementation.

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/ibm-products/blob/main/.github/CODE_OF_CONDUCT.md)
- [x] I checked the [current issues](https://github.com/carbon-design-system/ibm-products/issues) for duplicate problems

### Acceptance criteria

- [ ] Identify cases of `OperationalTag`
  - [ ] Add `aria-expanded`

This should account for both `TagSet` (`TagSetOverflow`) and `TagOverflow`, but there may be additional instances.
